### PR TITLE
added identify the problem episode for review, Dec2024. Changed to Rm…

### PR DESCRIPTION
**Pull request by peterlaurin**
…d to run R.

### CODE REVIEW

**Review by xortizross — APPROVED** [review]
I'm not reviewing the actual content, I'm just approving to see the rendered version :)

File: episodes/3-identify-the-problem.Rmd [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+
+ggplot(krats, aes(date, fill=plot_type)) + 
+geom_histogram() +
+facet_wrap(~species)
+
+```
+
+This looks much better, and is easier to see the trends over time as well. Note our x axis still shows bins with year labelings, but the continuous spread of our data over these bins shows that dates are treated more continuously and fall more continuously within histogram bins.
+
+:::::::: callout
```
Comment (kaijagahm) [inline episodes/3-identify-the-problem.Rmd:229]: In general I think this is a good callout, but I'm not entirely following the connection to reproducible examples. Can you explain what you meant by that? Why is this example more relevant to reproducible examples than the other examples in this episode?

File: episodes/3-identify-the-problem.Rmd [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+::::::::
+
+
+
+
+In general, we need to isolate the specific areas of code causing the bug or problem. There is no general rule of thumb as to how large this needs to be, but in general, think about what we would want to include in a reprex. Any early lines which we know run correctly and as intended may not need to be included, and we should seek to isolate the problem area as much as we can to make it understandable to others. 
+
+Let's add to our list of steps for identifying the problem: 
+
+0. Identify the problem area -- add print statements immediately upstream or downstream of problem areas, step into functions, and see whether any intermediate output can be further isolated.
```
Comment (kaijagahm) [inline episodes/3-identify-the-problem.Rmd:240]: The audience probably won't know what a "print statement" is or what it means to "step into functions". Do you think it's within the scope of this episode to cover those techniques?

File: episodes/3-identify-the-problem.Rmd [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+  ggplot(aes(x = date, fill = plot_type)) +
+  geom_histogram()+
+  facet_wrap(~species, ncol = 1)+ 
+  theme_bw()+
+  scale_fill_viridis_d(option = "plasma")+
+  geom_vline(aes(xintercept = lubridate::ymd("1988-01-01")), col = "dodgerblue")
+
+```
+
+It looks like the study change helped to reduce merriami sightings in the Rodent and Short-term Krat exclosures.
```
Comment (kaijagahm) [inline episodes/3-identify-the-problem.Rmd:276]: I suggest italicizing "merriami"

File: episodes/3-identify-the-problem.Rmd [hunk 4]
```diff
@@ -0,0 +1,279 @@
+
+---
+title: "Identify the problem and make a plan"
+teaching: 0
```
Comment (kaijagahm) [inline episodes/3-identify-the-problem.Rmd:4]: Need to update the time estimates for teaching and exercises here.

File: episodes/3-identify-the-problem.Rmd [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+:::::::: callout 
+One aspect we can see with this exercise above is that by setting up a reproducible example, we can isolate the problem with data rather than simply asking a proximal problem (i.e. 'how can i change my plot to look like so'). This allows helpers and you to directly improve your code, but also allows the community to help in identifying the problem. You don't always need to  understand what exact lines of code or function calls are going wrong in order to get help!
+::::::::
+
+
+## Summary
+
+In general, we need to isolate the specific areas of code causing the bug or problem. There is no general rule of thumb as to how large this needs to be, but in general, think about what we would want to include in a reprex. Any early lines which we know run correctly and as intended may not need to be included, and we should seek to isolate the problem area as much as we can to make it understandable to others.
```
Comment (xortizross) [inline episodes/3-identify-the-problem.Rmd:236]: I'm not sure that I am following...

File: episodes/3-identify-the-problem.Rmd [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+
+Let's add to our list of steps for identifying the problem: 
+
+0. Identify the problem area -- add print statements immediately upstream or downstream of problem areas, step into functions, and see whether any intermediate output can be further isolated. 
+
+1. Read each part of the error or warning message (if applicable), to see if we can immediately interpret and act on any.  
+
+2. Pulling up the R Documentation for any function calls causing the error (which may be specific to a package, such as with ggplot).
+
+3. Reading through the documentation's Description, Usage, Arguments, Details and Examples entries for greater insight into our error.
```
Comment (xortizross) [inline episodes/3-identify-the-problem.Rmd:246]: I feel like 2 and 3 cane just be one?

File: episodes/3-identify-the-problem.Rmd [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+
+2. Pulling up the R Documentation for any function calls causing the error (which may be specific to a package, such as with ggplot).
+
+3. Reading through the documentation's Description, Usage, Arguments, Details and Examples entries for greater insight into our error. 
+
+4. Copying and pasting the error message into a search engine for more interpretable explanations.
+
+And, when all else fails, preparing our code into a reproducible example for expert help.
+
+Whereas before we had a list of steps for addressing spot problems arising in one or two lines, we can now organize identifying the problem into a more organizational workflow. Any step in the above that helps us identify the specific areas or aspects of our code that are failing in particular, we can zoom in on and restart the checklist. We can stop as soon as we don't understand anymore how our code fails, at which point we can excise that area for further help.
```
Comment (xortizross) [inline episodes/3-identify-the-problem.Rmd:252]: Maybe flesh this out a little more?

**Review by xortizross — COMMENTED** [review]
Just some thoughts

**Pull request merged**